app/models/news_item.rb

```diff
@@ -1,7 +1,10 @@
 class NewsItem < ActiveRecord::Base
   extend Enumerize
-  enumerize :state, in: [:draft, :archived, :published]
+  AVAILABLE_STATES = %w(draft archived published)
+  enumerize :state, in: AVAILABLE_STATES
   scope :published, -> { where(state: :published).order(:published_at) }
-  validates :published_at, presence: true, if: -> { state.published? } 
-  validates_presence_of :title, :state
+  validates :published_at, presence: true, if: -> { state.try(:published?) }
+  validates :title, presence: true, length: { maximum: 255 }
+  validates :state, presence: true, inclusion: { in: AVAILABLE_STATES }
+  validates :body, presence: true, length: { maximum: 4000 }
```

Comment:
In my experience, some databases support up to 4000 characters for "text" columns (like "news_items"."body"), so I just did that reflexively thinking it was good enough a limit for now (I didn't bother to check what the limits are in this project's postgres, but hey 4000 is pretty high, no?! :) )